src/main/java/com/google/sps/servlets/articlesServlet.java

```diff
@@ -95,10 +126,46 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     }
 
     String articles = strBuf.toString();
+
+    Query query = new Query("Article");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    // Clears datastore before updating the results.
+    for (Entity entity : results.asIterable()) {
+      Key articleEntityKey = KeyFactory.createKey("Article", entity.getKey().getId());
+      datastore.delete(articleEntityKey);
+    }
+
+    // Formats data using the aritcle entity to be stored in the database.
+    JSONObject obj = new JSONObject(articles);
+    JSONArray articleData = obj.getJSONArray("articles");
+    int numArticles = articleData.length();
+
+    for (int i = 0; i < numArticles; ++i) {
+      JSONObject currentArticle = articleData.getJSONObject(i);
+      Entity articleEntity = new Entity("Article");
+
+      articleEntity.setProperty("source", currentArticle.getJSONObject("source").getString("name"));
+      articleEntity.setProperty("author", currentArticle.getString("author"));
+      articleEntity.setProperty("title", currentArticle.getString("title"));
+      articleEntity.setProperty("description", currentArticle.getString("description"));
+      articleEntity.setProperty("url", currentArticle.getString("url"));
+      articleEntity.setProperty("publishedAt", currentArticle.getString("publishedAt"));
+
+      datastore.put(articleEntity);
+    }
+    
     Gson gson = new Gson();
     String json = gson.toJson(articles);
 
     response.setContentType("application/json");
     response.getWriter().println(json);
   }
+
+  
+  @Override
+  public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    doGet(request, response);
```

Comment:
Why did you merge your doGet and doPut? Usually they should do different things.

Reply:
Yes! I just figured out a way to just merge them, so will implement that now, commit it, then merge to master. :)

Reply:
I was wondering why they should be merged. 

Usually get means the client is getting information from the server, and put means the client wants to store some information in server (and maybe get some results back), so they should serve different purposes and shouldn't be merged.

Reply:
If you are talking about remove one of the put and get, and keep the other one, I think it is fine.

Reply:
ok! Thank you!